rspace/src/main/scala/coop/rchain/rspace/history/HistoryStore.scala

```diff
@@ -34,8 +36,25 @@ object HistoryStoreInstances {
 
     override def get(key: Blake2b256Hash): F[Trie] =
       for {
-        maybeBytes <- store.get(key)
-        result     <- maybeBytes.traverse(bytes => Trie.codecTrie.decode(bytes).get)
+        bytesOpt <- store.get(key)
+        result <- bytesOpt.traverse(
+                   bytes =>
+                     for {
+                       decodeResult <- Trie.codecTrie.decode(bytes).pure[F]
+                       data <- decodeResult match {
+                                case Attempt.Successful(res) =>
+                                  Sync[F].pure(res)
+                                case Attempt.Failure(err) =>
+                                  Sync[F]
+                                    .raiseError[DecodeResult[Trie]](
+                                      new Exception(
+                                        s"Key: ${Base16.encode(key.bytes.toArray)}\nValue: ${Base16
+                                          .encode(bytes.toByteArray)}\n${err.toString()}"
+                                      )
+                                    )
+                              }
```

Comment:
The work of converting `Attempt` to Applicative error is already done in `decode(..).get` so here we can just "hook" on the error to display additional info. `onError` does not handle the error.
```scala
result <- maybeBytes.traverse { bytes =>
            Trie.codecTrie.decode(bytes).get.onError {
              case ex: Throwable =>
                val errMsg =
                  s"Key: ${Base16.encode(key.bytes.toArray)}\nValue: ${bytes.toHex}"
                Log[F].error(errMsg, ex)
            }
          }
```
Can you also try to implement retry logic so we can see if very next get will have the same error?